### PR TITLE
Implementation for persisting cloud events in events table

### PR DESCRIPTION
**Pull request by acn-sbuad**
## Description

Service, repository and DB are now ready to support external events. 
Also made changes to ensure app events are stored in both tables. 

## Related Issue(s)
- #156

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [x] Manual testing done (required)
~~- [ ] Relevant automated test added (if you find this hard, leave it and we'll help out)~~ will be handled when controller in place
- [x] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

File: src/Events/Migration/v0.21/01-setup-table.sql [hunk 1]
```diff
@@ -0,0 +1,6 @@
+CREATE TABLE IF NOT EXISTS events.events (
+    sequenceno BIGSERIAL,
+    cloudevent JSONB NOT NULL,
+    registeredtime timestamptz default (now() at time zone 'utc'),
+    CONSTRAINT events_pkey PRIMARY KEY (sequenceno)
```
Comment (SandGrainOne) [inline src/Events/Migration/v0.21/01-setup-table.sql:5]: Is this table now using the same sequence as the app events table?

Comment (acn-sbuad) [reply src/Events/Migration/v0.21/01-setup-table.sql:5]: no, the contraint was renamed for events_app table

File: src/Events/Repository/CloudEventRepository.cs [hunk 2]
```diff
@@ -49,6 +51,21 @@ public async Task CreateAppEvent(CloudEvent cloudEvent)
             pgcom.Parameters.Add(new NpgsqlParameter("cloudevent", cloudEvent.Serialize()) { Direction = System.Data.ParameterDirection.Input });
 
             await pgcom.ExecuteNonQueryAsync();
+
+            await using NpgsqlCommand pgcom2 = new NpgsqlCommand(string.Format(insertEventSql, serializedCloudEvent), conn);
```
Comment (SandGrainOne) [inline src/Events/Repository/CloudEventRepository.cs:55]: We should avoid building a SQL statement with any form of string manipulation. Always use parameters. 

When it comes to parameters we've usually named them, but @vgr has found that using ordering is best. Not sure how much it matters though.

Comment (b-brodie-digdir) [reply src/Events/Repository/CloudEventRepository.cs:55]: I find named parameters to be more readable and safer when refactoring. In this case, I would expect the performance difference should be non-measurable.

Comment (acn-sbuad) [reply src/Events/Repository/CloudEventRepository.cs]: Given that it's only a single parameter and we have the full sql statement available I'm going with the numbered parameter here, the property name should be telling enough w.r.t readability / refactoring

File: src/Events/Repository/CloudEventRepository.cs [hunk 3]
```diff
@@ -49,6 +51,33 @@ public async Task CreateAppEvent(CloudEvent cloudEvent)
             pgcom.Parameters.Add(new NpgsqlParameter("cloudevent", cloudEvent.Serialize()) { Direction = System.Data.ParameterDirection.Input });
 
             await pgcom.ExecuteNonQueryAsync();
+
+            await using NpgsqlCommand pgcom2 = new NpgsqlCommand(insertEventSql, conn)
+            {
+                Parameters =
```
Comment (SandGrainOne) [inline src/Events/Repository/CloudEventRepository.cs:57]: Didn't know this syntax was an option. The Parameters property is read only. What is the result of this syntax? Is the new parameter added to existing collection or does it create a new collection?

Comment (SandGrainOne) [reply src/Events/Repository/CloudEventRepository.cs:57]: Fun stuff:
https://sharplab.io/#v2:EYLgtghglgdgNAExAagD4AEBMBGAsAKHQAYACdbAFgG4CD0BmMzEgYRIG8CTuzH0KSAWQAUASg5ceUlgHsANnICmAYwAuUGTAAqigM6rFAJxIH9RkgF4SMRQHcxnfFOc8AMlH2WSkl1Me+AkgAiU1Ugn0CAXwipSJonHmiE7gjYA0MYCDkmVnklNQ1tPXSJZOcGEnd9AB5yIgA+EgB9OQ9VeICKqtVa7AbKto4SAHNFdpIkzsZZBRV1TR0zQzFSwJ4WwasbWwGauvqxDt9JxIJIoA===

Comment (SandGrainOne) [reply src/Events/Repository/CloudEventRepository.cs:57]: In other words. Elements are added

**Review by b-brodie-digdir — DISMISSED** [review]

File: src/Events/Repository/CloudEventRepository.cs [hunk 4]
```diff
@@ -35,11 +36,12 @@ public CloudEventRepository(IOptions<PostgreSQLSettings> postgresSettings)
         }
 
         /// <inheritdoc/>
-        public async Task CreateAppEvent(CloudEvent cloudEvent)
+        public async Task CreateAppEvent(CloudEvent cloudEvent, string serializedCloudEvent)
         {
             await using NpgsqlConnection conn = new NpgsqlConnection(_connectionString);
             await conn.OpenAsync();
 
+            await using var transaction = await conn.BeginTransactionAsync();
```
[the remark this answers is not in the document]
Comment (acn-sbuad) [reply src/Events/Repository/CloudEventRepository.cs:44]: Might just be a question of notation here. The `using` without the curly braces means the block runs to the end of the function. So I believe this is already in place.

[the remark this answers is not in the document]
Comment (SandGrainOne) [reply src/Events/Repository/CloudEventRepository.cs:44]: The transaction class does a rollback during dispose if not completed in some other way.
https://github.com/npgsql/npgsql/blob/docs/src/Npgsql/NpgsqlTransaction.cs/#L366

**Review by b-brodie-digdir — APPROVED** [review]

**Review by SandGrainOne — APPROVED** [review]

**Pull request merged**